### PR TITLE
Garbage collection

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and comment / commit if anything is wrong.

### CODE REVIEW

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 1]
```diff
@@ -1,63 +1,63 @@
...
-    - Other functions on the current chain of nested calls, their local variables and parameters.
-    - Global variables.
-    - (there are some other, internal ones as well)
+    - تابعی که در حال اجرا باشد، متغیرهای محلی و پارامترهای آن.
+    - تابع‌های دیگر در زنجیره‌ی کنونیِ صدازدن های تو در تو، متغیرهای محلی و پارامترهای آن.
+    - متغیرهای global.
+    - (چند مورد دیگر هم هستند، همچنین موردهای داخلی)
 
-    These values are called *roots*.
+    این مقدارها *ریشه* نامیده می‌شوند.
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:22]:
```suggestion
    این مقدارها *ریشه‌ها* نامیده می‌شوند.
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 2]
```diff
@@ -1,63 +1,63 @@
...
 
-## A simple example
+## یک مثال ساده
 
-Here's the simplest example:
+اینجا ساده‌ترین مثال را داریم:
 
 ```js
-// user has a reference to the object
+// به شیء رجوع می‌کند user
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:35]:
```suggestion
// یک ارجاع به شیء دارد user
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 3]
```diff
@@ -1,63 +1,63 @@
...
 
-## Two references
+## دو مرجع
 
-Now let's imagine we copied the reference from `user` to `admin`:
+حال بیایید تصور کنیم که مرجع را از `user` در `admin` کپی کردیم:
 
 ```js
-// user has a reference to the object
+// دارای مرجعی به شیء است user
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:60]: Please use the same translations for the exact same phrases.
```suggestion
// یک ارجاع به شیء دارد user
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 4]
```diff
@@ -115,98 +115,98 @@ delete family.mother.husband;
...
-But if we delete both, then we can see that John has no incoming reference any more:
+اما اگر ما هر دو را حذف کنیم، آن گاه می‌بینیم که John دیگر هیچ مرجع ورودی ندارد:
 
 ![](family-no-father.svg)
 
-Outgoing references do not matter. Only incoming ones can make an object reachable. So, John is now unreachable and will be removed from the memory with all its data that also became unaccessible.
+مرجع‌های خروجی مهم نیستند. تنها مرجع‌های ورودی می‌توانند یک شیء را قابل دسترس کنند. پس John حالا غیر قابل دسترس شده است و همراه با تمام داده‌اش که آنها هم غیر قابل دسترس شده اند، از حافظه پاک می‌شود.
 
-After garbage collection:
+بعد از جمع‌آوری زباله:
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:126]:
```suggestion
بعد از بازیافت حافظه:
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 5]
```diff
@@ -115,98 +115,98 @@ delete family.mother.husband;
...
+- مرجع بودن با قابل دسترس بودن (از یک ریشه) یکسان نیست : یک دسته‌ی شیءهای بهم پیوسته می‌توانند به طور کامل غیر قابل دسترس شوند.
 
-Modern engines implement advanced algorithms of garbage collection.
+موتورهای مدرن الگوریتم‌های پیشرفته‌ی جمع‌آوری زباله را پیاده‌سازی می‌کنند.
 
-A general book "The Garbage Collection Handbook: The Art of Automatic Memory Management" (R. Jones et al) covers some of them.
+کتاب کلی "The Garbage Collection Handbook: The Art of Automatic Memory Management" (R. Jones و بقیه افراد) بعضی از آنها را پوشش می‌دهد.
 
-If you are familiar with low-level programming, the more detailed information about V8 garbage collector is in the article [A tour of V8: Garbage Collection](http://jayconrod.com/posts/55/a-tour-of-v8-garbage-collection).
+اگر شما با برنامه‌نویسی سطح پایین آشنایی دارید، اطلاعاتی با جزییات درباره زباله جمع‌کن V8 در این مقاله است [A tour of V8: Garbage Collection](http://jayconrod.com/posts/55/a-tour-of-v8-garbage-collection).
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:208]:
```suggestion
اگر شما با برنامه‌نویسی سطح پایین آشنایی دارید، اطلاعاتی با جزییات درباره زباله‌روبی V8 در این مقاله است [A tour of V8: Garbage Collection](http://jayconrod.com/posts/55/a-tour-of-v8-garbage-collection).
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 6]
```diff
@@ -1,63 +1,63 @@
-# Garbage collection
+# جمع‌آوری زباله
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:1]:
```suggestion
# بازیافت حافظه
```

Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:1]:
```suggestion
# زباله‌روبی
```

File: 1-js/04-object-basics/03-garbage-collection/article.md [hunk 7]
```diff
@@ -1,63 +1,63 @@
...
 
-2. Any other value is considered reachable if it's reachable from a root by a reference or by a chain of references.
+2. هر مقدار دیگری قابل دسترس فرض می‌شود اگر از یک ریشه توسط یک مرجع یا زنجیره‎ای از مراجع قابل دسترس باشد.
 
-    For instance, if there's an object in a global variable, and that object has a property referencing another object, *that* object is considered reachable. And those that it references are also reachable. Detailed examples to follow.
+    برای مثال، اگر یک شیء در متغیری global وجود داشته باشد، و آن شیء یک ویژگی داشته باشد که به شیءای دیگر رجوع می‌کند، *آن* شیء قابل دسترس فرض می‌شود. و آنهایی که این شیء به آنها رجوع می‌کند هم قابل دسترس هستند. مثال‌های دارای جزئیات در ادامه آمده است.
 
-There's a background process in the JavaScript engine that is called [garbage collector](https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)). It monitors all objects and removes those that have become unreachable.
+یک فرایند پشت پرده در موتور جاوااسکریپت وجود دارد به نام [زباله جمع‌کن](https://fa.wikipedia.org/wiki/بازیافت_حافظه). این فرایند تمام شیءها را زیر نظر می‌گیرد و آنهایی که غیر قابل دسترس شده‌اند را پاک می‌کند.
```
Comment (mahdyar) [inline 1-js/04-object-basics/03-garbage-collection/article.md:28]:
```suggestion
یک فرایند پشت پرده در موتور جاوااسکریپت وجود دارد به نام [زباله‌روبی](https://fa.wikipedia.org/wiki/بازیافت_حافظه). این فرایند تمام شیءها را زیر نظر می‌گیرد و آنهایی که غیر قابل دسترس شده‌اند را پاک می‌کند.
```

Comment (mahdyar) [reply 1-js/04-object-basics/03-garbage-collection/article.md:28]:
```suggestion
یک فرایند پشت پرده در موتور جاوااسکریپت وجود دارد به نام [بازیافت حافظه](https://fa.wikipedia.org/wiki/بازیافت_حافظه). این فرایند تمام شیءها را زیر نظر می‌گیرد و آنهایی که غیر قابل دسترس شده‌اند را پاک می‌کند.
```

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Comment by mahdiHashemi14** [comment]
Thanks for your review, @mahdyar
/done

**Review by mahdyar — APPROVED** [review]
Thank you, @mahdiHashemi14!

**Pull request merged**